### PR TITLE
chore(deps): Bump Rust to 1.75.0

### PR DESCRIPTION
**Pull request by neuronull**
Apologies for not separating the clippy fixes by rust version- will keep that in mind in the future.

There weren't really many changes that required decisions. Closest being a couple of field / struct name changes.

### CODE REVIEW

**Comment by neuronull** [comment]
>  We should be able to get rid of at least most of our uses of async_trait with this update. Do we want to do that in a follow up PR?

I can create a follow-up PR for that, thanks for flagging.

File: lib/vector-buffers/src/variants/disk_v2/reader.rs [hunk 1]
```diff
@@ -394,7 +394,7 @@ where
 
 /// Reads records from the buffer.
 #[derive(Debug)]
-pub struct Reader<T, FS>
+pub struct BufferReader<T, FS>
```
[the remark this answers is not in the document]
Comment (neuronull) [reply lib/vector-buffers/src/variants/disk_v2/reader.rs:397]: The clippy lint flagged the struct member name `reader` as being the same as the struct name.
I couldn't come up with a meaningful name for the member `reader` that didn't also end or begin with `reader`, so a more descriptive struct name is the solution I came up with. 

https://rust-lang.github.io/rust-clippy/master/index.html#/struct_field_names

File: lib/vector-buffers/src/variants/disk_v2/ledger.rs [hunk 2]
```diff
@@ -92,16 +92,16 @@ pub enum LedgerLoadCreateError {
 pub struct LedgerState {
     /// Next record ID to use when writing a record.
     #[with(Atomic)]
-    writer_next_record_id: AtomicU64,
+    writer_next_record: AtomicU64,
```
[the remark this answers is not in the document]
Comment (neuronull) [reply lib/vector-buffers/src/variants/disk_v2/ledger.rs:95]: yes, there is a lint now for when all of the field names have the same prefix/suffix.

https://rust-lang.github.io/rust-clippy/master/index.html#/struct_field_names

[the remark this answers is not in the document]
Comment (dsmith3197) [reply lib/vector-buffers/src/variants/disk_v2/ledger.rs:95]: Makes sense, thanks!

**Comment by neuronull** [comment]
> > We should be able to get rid of at least most of our uses of async_trait with this update. Do we want to do that in a follow up PR?
> 
> I can create a follow-up PR for that, thanks for flagging.

Actually, I started looking into this more and it appears this limitation:

https://blog.rust-lang.org/2023/12/21/async-fn-rpit-in-traits.html#async-fn-in-public-traits

Highly affects us.

Basically all of our traits that are using `async_trait`, are publicly defined. I tried changing one and got the warning that is described in that blog post.
So, I think we might be better off waiting for this to get fully ironed out.

cc @StephenWakely

**Review by bruceg — APPROVED** [review]

**Comment by bruceg** [comment]
> We should be able to get rid of at least most of our uses of `async_trait` with this update. Do we want to do that in a follow up PR?

Only if we also increase the MSRV as well, which we haven't been doing.

**Review by dsmith3197 — APPROVED** [review]

**Comment by StephenWakely** [comment]
> Basically all of our traits that are using `async_trait`, are publicly defined.

Darn it. I was hoping that we would be able to get away with that given we aren't really a publicly consumed library..

**Pull request merged**